### DISCUSSION TITLE
图片大小是不是有点压不住了🤔

### DISCUSSION THREAD

**Discussion by kane-le** [Cute-Dress/Dress#95] [畅聊]
整个项目4.4GB了，看单张照片还有达到4-5M的，现在已经不限制图片大小了嘛

**Comment by CuteDress** [comment]
原则上嘛，还是要限制的，至于单张超出大小的有很多是这个项目（我开始维护）之前留下的，也没办法了

**Comment by Shiroiame-Kusu** [comment]
pull下来要一万年，sad:(

**Reply by Young-Lord** [reply]
其实可以 GitHub Codespces 或者 `git clone <URL> --no-checkout <directory> --depth 1`

**Comment by Young-Lord** [comment]
话说文件只要曾经存在过就会占空间（比如 push 后压缩再 push，或者 push 后删除），并且删掉文件也不会释放空间（除非 rebase，但那样成本太高了）
这就是git.jpg

**Comment by rustyllh** [comment]
多来点大提交塞满这个小仓库😋

**Comment by Foxie-404** [comment]
要不先pull下来将超出一定阈值的文件统一压缩，rebase root 后重新上传？

**Comment by hnrobert** [comment]
其实可以直接用 GitHub Codespaces

<img width="499" height="463" alt="image" src="https://github.com/user-attachments/assets/6c9db0ee-8229-45dc-a4ab-b82704b048a6" />

这样就可以免去每次需要 pull 到本地的烦恼

然后关于体积我在写一个优化这部分的 action，包括 `squash merge` 时自动去除 EXIF 信息的功能, currently wip.

<https://github.com/hnrobert/git-image-preprocessor>

**Comment by hnrobert** [comment]
然后关于巨大的历史 commit 应该也是有办法可以 fix 的，只是如果这样很难避免的是 所有人的 fork 都会炸

到时候看你们意愿了，这个放到别的项目也会是个有用的东西
